### ISSUE TITLE
Train with Spark generates ExecutorLostFailure

### ISSUE THREAD

**Issue by vzamboni** [deeplearning4j/deeplearning4j#1864]
I'm training a Convolutional Neural Network on a small dataset of images (15 classes, 70imgs per class).
I tried with 2 different clusters:
- 7 nodes, 8gb, 8 cores, physical nodes with Spark 1.6.0 standalone 
- 3 nodes, 60gb, 8 cores, AWS with Spark 1.6.2 standalone

And I'm always facing ExecutorLostFailure:

> ERROR TaskSchedulerImpl: Lost executor 36 on < ip-of-the-node >: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.

Training keeps going but a lot of executors get disassociated till it crashes. Before it crashes the training is performing good and it reaches 40-50% f1-score.

From the Spark UI the situation is [this](https://s32.postimg.org/7r5k9urv9/exec.png)

my code is [here](https://gist.github.com/vzamboni/e8997b4a01ea1a5dc262fe7a5dc3473c)

I'm using version 0.4.0 but I had the same problem with previous version.
I compiled deepleraning4j, nd4j and canova with scala 2.10 and I'm using spark 1.6

Launch configuration is:

> spark-submit --conf spark.driver.maxResultSize=2g --jars  --master -my-spark-master- --driver-memory 4g --executor-memory 50g --class SparkConvolutional my-jar 

Dependencies of the projects are [here](https://gist.github.com/vzamboni/da1be46d8c46b259a6fccdc6c2fded1e)
# UPDATE

I tried with the examples in the dl4j-spark-cdh5-examples repo (MnistExample and GravesLSTMCharModellingExample) in the 7 nodes cluster with scala 2.11 and building with maven and I'm facing the same problem.

MnistExample reaches the end (epoch 4) but throws some ExecutorLostFailure during the training.
GravesLSTMCharModellingExample does not finish and crashes after 50 minutes and a lot of ExecutorLostFailure.

In both the executions I'm having a lot of 

> WARN TaskSetManager: Stage 0 contains a task of very large size (3898 KB). The maximum recommended task size is 100 KB
> which I noticed increasing with dataset size increasing.

I have both full logs of [MnistExample](https://gist.github.com/vzamboni/cc3816341a1dfb0304aef0084bbf5364) (example of ExecutorLostFailure at line 903) and [GravesLSTMCharModellingExample ](https://gist.github.com/vzamboni/a733aefe9a0bc7820afb4313b397d94a) (example of ExecutorLostFailure at line 132)

**Comment by vzamboni** [comment]
As a workaround I had to decrease the amount of memory given to the executors. In a 7 nodes x 6g cluster I start the program with:
> --driver-memory 2g --executor-memory 2g

Decreasing the memory dedicated to the executors I have the program performing good now. 

I am now trying with a bigger network, (I used [this](https://github.com/deeplearning4j/hadoop-summit-scenes/blob/master/src/main/java/org/deeplearning4j/Train.java) with the same parameter used in the example) and I got an ExecutorLostFailure after 13 minutes, but the situation is far better now.
